src/lib/storage/chunk_encoder.cpp

```diff
@@ -54,6 +54,7 @@ std::shared_ptr<AbstractSegment> ChunkEncoder::encode_segment(const std::shared_
     if (encoding_spec.encoding_type == EncodingType::Unencoded) {
       pmr_vector<ColumnDataType> values;
       pmr_vector<bool> null_values;
+      auto nullable = bool{false};
```

Comment:
I think the term `nullable` is confusing here. Maybe `contains_nulls`?

A segment without NULLs can still be nullable (let's ignore MVCC/append-only). At least we usually use these terms this way.